pkg/controller/pxcrestore/util.go

```diff
@@ -0,0 +1,105 @@
+package pxcrestore
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	api "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+)
+
+func getBackup(ctx context.Context, cl client.Client, cr *api.PerconaXtraDBClusterRestore) (*api.PerconaXtraDBClusterBackup, error) {
+	if cr.Spec.BackupSource != nil {
+		status := cr.Spec.BackupSource.DeepCopy()
+		status.State = api.BackupSucceeded
+		status.CompletedAt = nil
+		status.LastScheduled = nil
+		return &api.PerconaXtraDBClusterBackup{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      cr.Name,
+				Namespace: cr.Namespace,
+			},
+			Spec: api.PXCBackupSpec{
+				PXCCluster:  cr.Spec.PXCCluster,
+				StorageName: cr.Spec.BackupSource.StorageName,
+			},
+			Status: *status,
+		}, nil
+	}
+
+	bcp := &api.PerconaXtraDBClusterBackup{}
+	err := cl.Get(ctx, types.NamespacedName{Name: cr.Spec.BackupName, Namespace: cr.Namespace}, bcp)
+	if err != nil {
+		err = errors.Wrapf(err, "get backup %s", cr.Spec.BackupName)
+		return bcp, err
+	}
+	if bcp.Status.State != api.BackupSucceeded {
+		err = errors.Errorf("backup %s didn't finished yet, current state: %s", bcp.Name, bcp.Status.State)
+		return bcp, err
+	}
+
+	return bcp, nil
+}
+
+func setStatus(ctx context.Context, cl client.Client, cr *api.PerconaXtraDBClusterRestore, state api.BcpRestoreStates, comments string) error {
+	cr.Status.State = state
+	switch state {
+	case api.RestoreSucceeded:
+		tm := metav1.NewTime(time.Now())
+		cr.Status.CompletedAt = &tm
+	}
+
+	cr.Status.Comments = comments
+
+	err := cl.Status().Update(ctx, cr)
+	if err != nil {
+		return errors.Wrap(err, "send update")
+	}
+
+	return nil
+}
+
+func isOtherRestoreInProgress(ctx context.Context, cl client.Client, cr *api.PerconaXtraDBClusterRestore) (*api.PerconaXtraDBClusterRestore, error) {
+	rJobsList := &api.PerconaXtraDBClusterRestoreList{}
+	err := cl.List(
+		ctx,
+		rJobsList,
+		&client.ListOptions{
+			Namespace: cr.Namespace,
+		},
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "get restore jobs list")
+	}
```

Comment:
We can write this like that:

```
        rJobsList := &api.PerconaXtraDBClusterRestoreList{}
	if err := cl.List(
		ctx,
		rJobsList,
		&client.ListOptions{
			Namespace: cr.Namespace,
		},
	); err != nil {
		return nil, errors.Wrap(err, "get restore jobs list")
	}
```

Since the same error is not used after that context

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/1610/commits/f8b07591d3e50248e70175a08f9b25356ea3f93d